VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Structs/StructWithoutElementaryMethodsOverridden/StructWithoutElementaryMethodsOverriddenCodeFix.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.Structs.StructWithoutElementaryMethodsOverridden
+{
+    [ExportCodeFixProvider(DiagnosticId.StructWithoutElementaryMethodsOverridden + "CF", LanguageNames.CSharp), Shared]
+    public class StructWithoutElementaryMethodsOverriddenCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(StructWithoutElementaryMethodsOverriddenAnalyzer.Rule.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            string implementEquals;
+            string implementGetHashCode;
+            string implementToString;
+
+            diagnostic.Properties.TryGetValue("IsEqualsImplemented", out implementEquals);
+            diagnostic.Properties.TryGetValue("IsGetHashCodeImplemented", out implementGetHashCode);
+            diagnostic.Properties.TryGetValue("IsToStringImplemented", out implementToString);
+
+            var statement = root.FindNode(diagnosticSpan);
+
+            context.RegisterCodeFix(CodeAction.Create(VSDiagnosticsResources.ReplaceEmptyStringWithStringDotEmptyCodeFixTitle,
+                    x => AddMissingMethodsAsync(context.Document, root, (StructDeclarationSyntax) statement,
+                            bool.Parse(implementEquals), bool.Parse(implementGetHashCode), bool.Parse(implementToString)),
+                    StructWithoutElementaryMethodsOverriddenAnalyzer.Rule.Id), diagnostic);
+        }
+
+        private Task<Document> AddMissingMethodsAsync(Document document, SyntaxNode root,
+            StructDeclarationSyntax statement, bool implementEquals, bool implementGetHashCode,
+            bool implementToString)
+        {
+            var newStatement = statement;
+
+            if (!implementEquals)
+            {
+                newStatement = newStatement.AddMembers(GetEqualsMethod());
+            }
+
+            if (!implementGetHashCode)
+            {
+                newStatement = newStatement.AddMembers(GetGetHashCodeMethod());
+            }
+
+            if (!implementToString)
+            {
+                newStatement = newStatement.AddMembers(GetToStringMethod());
+            }
+
+            var newRoot = root.ReplaceNode(statement, newStatement);
+            return Task.FromResult(document.WithSyntaxRoot(newRoot));
+        }
+
+        private MethodDeclarationSyntax GetEqualsMethod()
```

Comment:
Can you make three static fields of type `MethodDeclarationSyntax` instead and use the static constructor of `StructWithoutElementaryMethodsOverriddenCodeFix` to assign these fields? That way they will only be calculated once (when the class is loaded) and every time the CodeFix is executed it can just return the `MethodDeclarationSyntax` it computed.